internal/controller/argocd.go

```diff
@@ -47,7 +47,7 @@ func (clctrl *ClusterController) InstallArgoCD() error {
 		}
 
 		var argoCDInstallPath string
-		if clctrl.CloudProvider == "digitalocean" || clctrl.CloudProvider == "aws" {
+		if clctrl.CloudProvider == "digitalocean" || clctrl.CloudProvider == "aws" || clctrl.CloudProvider == "civo" {
```

Comment:
Since this might start growing, what about...


```go
switch clctrl.CloudProvider {
  case "digitalocean", "aws", "civo", ...:
     // do the upgraded ones here

  default:
     // do it old fashion
}
```

Reply:
got it ;)

Reply:
and also isnt switch slower than if else?

Reply:
Let's try it!

Here's a Go benchmark:

```go
package main

import (
	"testing"
)

func compareWithSwitch(s string) string {
	switch s {
	case "apple":
		return "fruit"
	case "carrot":
		return "vegetable"
	case "banana":
		return "fruit"
	case "potato":
		return "vegetable"
	case "strawberry":
		return "fruit"
	case "tomato":
		return "fruit"
	default:
		return "unknown"
	}
}

func compareWithIfElse(s string) string {
	if s == "apple" {
		return "fruit"
	} else if s == "carrot" {
		return "vegetable"
	} else if s == "banana" {
		return "fruit"
	} else if s == "potato" {
		return "vegetable"
	} else if s == "strawberry" {
		return "fruit"
	} else if s == "tomato" {
		return "fruit"
	} else {
		return "unknown"
	}
}

func BenchmarkCompareWithSwitch(b *testing.B) {
	for i := 0; i < b.N; i++ {
		for j := 0; j < 10; j++ { // Increase the workload by repeating the operation
			_ = compareWithSwitch("carrot")
		}
	}
}

func BenchmarkCompareWithIfElse(b *testing.B) {
	for i := 0; i < b.N; i++ {
		for j := 0; j < 10; j++ { // Increase the workload by repeating the operation
			_ = compareWithIfElse("carrot")
		}
	}
}
```

If I run this a few times, I get:

```
$ go test -bench=. -benchmem 
goos: darwin
goarch: arm64
pkg: github.com/patrickdappollonio/temp-1dd4c79d
cpu: Apple M3 Pro
BenchmarkCompareWithSwitch-11           328892221                3.472 ns/op           0 B/op          0 allocs/op
BenchmarkCompareWithIfElse-11           336740256                3.687 ns/op           0 B/op          0 allocs/op
PASS
ok      github.com/patrickdappollonio/temp-1dd4c79d     3.296s
```

(On `ns/ops`, lower is better). It seems the switch is lightly faster by .215 nanoseconds 😆 while that could be important in a future grand scheme of things, the gains on readability far surpass the .215 per check IMHO!

Reply:
Absolutely, readability often outweighs the tiny performance gains in most cases. A clearer, more understandable codebase is usually more valuable in the long run, especially for maintenance and collaboration. Plus, the difference of .215 nanoseconds per check is so minuscule that it’s unlikely to have a noticeable impact on overall performance. Prioritizing code clarity is definitely a win for both current and future developers! 😄